src/components/restore-password/reset-password.component.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from "react";
+import { Formik, Form } from "formik";
+import { withTranslation } from "react-i18next";
+import { TextField } from "@material-ui/core";
+
+import history from "../../root/root.history";
+
+import UserService from "../../services/user.service";
+
+import "./restore-password.component.css";
+
+import { LOGIN_PAGE } from "../../root/root.constants";
+
+const userService = UserService.getInstance();
+
+class ResetPassword extends Component {
+  state = {
+    error: null,
+    loading: false,
+  };
+
+  handleResetPassword = (values) => {
+    this.setState({
+      loading: true,
+      error: null,
+    });
+    const url = window.location.hash.slice(window.location.hash.indexOf('?'));
+    const params = new URLSearchParams(url);
```

Comment:
Is there a better way extract query parameters from URL? I've tried `window.location.search` and it doesn't work because the URL contains a `#` in it.

Reply:
There isn't right now; however, I believe issue #343 will fix that. So it's the right way for now :D